### PR TITLE
Add CustomMetadata to top-level of pipeline spec

### PR DESCRIPTION
**Pull request by kevindelgado**
Move annotations out of service spec into top-level custom_metadata field

### CODE REVIEW

**Comment by kevindelgado** [comment]
closes #4598

**Comment by kevindelgado** [comment]
PTAL @jdoliner

**Comment by kevindelgado** [comment]
name updated, I believe I've updated the relevant docs, but because this is api breaking you might want to take a pass @svekars

File: doc/docs/master/reference/pipeline_spec.md [hunk 1]
```diff
@@ -12,6 +12,14 @@ create pipeline](pachctl/pachctl_create_pipeline.md) section.
...
   "description": string,
+  "metadata": {
+    "annotations": {
+        "foo": "bar"
+    },
+    "labels": {
+        "baz": "baq"
+    }
+  },
   "transform": {
```
Comment (svekars) [inline doc/docs/master/reference/pipeline_spec.md]: Need to add below after ### Description something like this (feel free to correct):

```
### Metadata

This parameter enables you to add metadata to your pipeline pods by using Kubernetes' `labels` and `annotations`. Labels help you to organize and keep track of your cluster objects by creating groups of pods based on the application they run, resources they use, or other parameters. Labels simplify the querying of Kubernetes objects and are handy in operations. 

Similarly to labels, you can add metadata through annotations. The difference is that you can specify any arbitrary metadata through annotations.

Both parameters require a key-value pair.  Do not confuse this parameter with `pod_patch` which adds metadata to the user container of the pipeline pod. For more information, see [Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) and [Kubernetes Annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) in the Kubernetes documentation. 
```

Comment (kevindelgado) [reply doc/docs/master/reference/pipeline_spec.md]: done

**Comment by kevindelgado** [comment]
@jdoliner feel free to take another pass, restore is now working

File: doc/docs/1.9.x/reference/pipeline_spec.md [hunk 2]
```diff
@@ -12,6 +12,14 @@ create pipeline](pachctl/pachctl_create_pipeline.md) section.
     "name": string
   },
   "description": string,
+  "metadata": {
+    "annotations": {
+        "foo": "bar"
```
Comment (jdoliner) [inline doc/docs/1.9.x/reference/pipeline_spec.md]:
```suggestion
        "annotation": string
```
This is the syntax we have throughout this doc, I'm not the biggest fan of it, but might as well keep it consistent until we're ready to change it all.

File: doc/docs/1.9.x/reference/pipeline_spec.md [hunk 3]
```diff
@@ -12,6 +12,14 @@ create pipeline](pachctl/pachctl_create_pipeline.md) section.
...
+    },
+    "labels": {
+        "baz": "baq"
```
Comment (jdoliner) [inline doc/docs/1.9.x/reference/pipeline_spec.md]:
```suggestion
        "label": string
```
Same here.

File: doc/docs/1.9.x/reference/pipeline_spec.md (hunk 2 above)
Comment (jdoliner) [reply doc/docs/1.9.x/reference/pipeline_spec.md]: Actually do we want to make these changes at all? This is in the 1.9.x version of the docs. And these features aren't available in 1.9.x releases.

File: doc/docs/master/reference/pipeline_spec.md [hunk 4]
```diff
@@ -12,6 +12,14 @@ create pipeline](pachctl/pachctl_create_pipeline.md) section.
     "name": string
   },
   "description": string,
+  "metadata": {
+    "annotations": {
+        "foo": "bar"
```
Comment (jdoliner) [inline doc/docs/master/reference/pipeline_spec.md]:
```suggestion
        "annotations": string
```

File: doc/docs/master/reference/pipeline_spec.md [hunk 5]
```diff
@@ -12,6 +12,14 @@ create pipeline](pachctl/pachctl_create_pipeline.md) section.
...
+    },
+    "labels": {
+        "baz": "baq"
```
Comment (jdoliner) [inline doc/docs/master/reference/pipeline_spec.md]:
```suggestion
        "label": string
```

File: doc/docs/1.9.x/reference/pipeline_spec.md (hunk 2 above)
Comment (kevindelgado) [reply doc/docs/1.9.x/reference/pipeline_spec.md]: nope, removed

**Comment by kevindelgado** [comment]
CI passing, do you want to take one final pass @jdoliner

**Pull request merged**